components/motor/gpio/basic.go

```diff
@@ -190,35 +195,34 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 
 	switch m.motorType {
 	case ABPwm, DirectionPwm:
-		if math.Abs(powerPct) <= 0.001 {
+		if pwmPower <= 0.001 {
 			return m.turnOff(ctx, extra)
 		}
 		pwmPin = m.PWM
 	case AB:
 		switch {
-		case powerPct >= 0.001:
+		case pwmPower >= 0.001:
 			pwmPin = m.B
 			if m.dirFlip {
 				pwmPin = m.A
 			}
-			powerPct = 1.0 - math.Abs(powerPct) // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
-		case powerPct <= -0.001:
+			pwmPower = 1.0 - pwmPower // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
+		case pwmPower <= -0.001:
 			pwmPin = m.A
 			if m.dirFlip {
 				pwmPin = m.B
 			}
-			powerPct = 1.0 - math.Abs(powerPct) // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
+			pwmPower = 1.0 - pwmPower // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
 		default:
 			return m.turnOff(ctx, extra)
 		}
 	}
 
-	powerPct = math.Max(math.Abs(powerPct), m.minPowerPct)
-	m.powerPct = powerPct
+	pwmPower = math.Max(pwmPower, m.minPowerPct)
```

Comment:
`m.powerPct` needs to capture the minPowerPct

Reply:
I'm going to move the power calculation up. New PR incoming.